### PR TITLE
Add Basic authentication and JWT based authentication support for GraphQL APIs

### PR DESCRIPTION
**Pull request by ChamodDamitha**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java [hunk 1]
```diff
@@ -253,6 +266,33 @@ public boolean validateScopes(String username, OpenAPI openAPI, MessageContext s
...
 
+    private String[] getUserRoles(String username) throws APISecurityException {
+        String[] userRoles;
+        String tenantDomain = MultitenantUtils.getTenantDomain(username);
+        if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            try {
+                userRoles = remoteUserStoreManagerServiceStub
+                        .getRoleListOfUser(MultitenantUtils.getTenantAwareUsername(username));
+            } catch (RemoteException | RemoteUserStoreManagerServiceUserStoreExceptionException e) {
+                throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR, e.getMessage());
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java]: Swallowing exception

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java [hunk 2]
```diff
@@ -253,6 +266,33 @@ public boolean validateScopes(String username, OpenAPI openAPI, MessageContext s
...
+                int tenantId = ServiceReferenceHolder.getInstance()
+                        .getRealmService().getTenantManager().getTenantId(tenantDomain);
+
+                UserStoreManager manager = ServiceReferenceHolder
+                        .getInstance().getRealmService()
+                        .getTenantUserRealm(tenantId).getUserStoreManager();
+
+                userRoles = manager.getRoleListOfUser(MultitenantUtils.getTenantAwareUsername(username));
+            } catch (UserStoreException e) {
+                throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR, e.getMessage());
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java]: Swallowing exception

Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java]: done

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java (hunk 1 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java]: done

**Pull request merged**